support-frontend/app/admin/settings/Switches.scala

```diff
@@ -1,21 +1,121 @@
+)
+
+case class SubscriptionsSwitches(
+  enableDigitalSubGifting: SwitchState = On,
+  useDotcomContactPage: SwitchState = On,
+  checkoutPostcodeLookup: SwitchState = On
+)
+
+case class RecaptchaSwitches(
+  enableRecaptchaBackend: SwitchState = On,
+  enableRecaptchaFrontend: SwitchState = On
+)
+
+case class OneOffPaymentMethodSwitches(
+  stripe: SwitchState = On,
+  stripeApplePay: SwitchState = On,
+  stripePaymentRequestButton: SwitchState = On,
+  payPal: SwitchState = On,
+  amazonPay: SwitchState = On
+)
+
+case class RecurringPaymentMethodSwitches(
+  stripe: SwitchState = On,
+  stripeApplePay: SwitchState = On,
+  stripePaymentRequestButton: SwitchState = On,
+  payPal: SwitchState = On,
+  directDebit: SwitchState = On,
+  existingCard: SwitchState = On,
+  existingDirectDebit: SwitchState = On,
+  amazonPay: SwitchState = Off,
+  sepa: SwitchState = Off
+)
 
 case class Switches(
-  oneOffPaymentMethods: PaymentMethodsSwitch,
-  recurringPaymentMethods: PaymentMethodsSwitch,
-  enableDigitalSubGifting: SwitchState,
-  useDotcomContactPage: Option[SwitchState],
-  enableRecaptchaBackend: SwitchState,
-  enableRecaptchaFrontend: SwitchState,
-  experiments: Map[String, ExperimentSwitch],
-  enableContributionsCampaign: SwitchState,
-  forceContributionsCampaign: SwitchState,
-  enableQuantumMetric: SwitchState
+  oneOffPaymentMethods: OneOffPaymentMethodSwitches,
+  recurringPaymentMethods: RecurringPaymentMethodSwitches,
+  subscriptionsSwitches: SubscriptionsSwitches,
+  featureSwitches: FeatureSwitches,
+  campaignSwitches: CampaignSwitches,
+  recaptchaSwitches: RecaptchaSwitches
 )
 
 object Switches {
-  implicit val switchesCodec: Codec[Switches] = deriveCodec
+  /**
+   * Transforms from e.g:
+   *  {
+   *    "oneOffPaymentMethods" : {
+   *      "description" : "Payment methods - one-off contributions",
+   *      "switches" : {
+   *        "stripe" : {
+   *          "description" : "Stripe - Credit/Debit card",
+   *          "state" : "On"
+   *        }
+   *      }
+   *    }
+   *  }
+   *  To:
+   *  {
+   *    "oneOffPaymentMethods" : {
+   *      "stripe": "On"
+   *    }
+   *  }
+   */
+  private def flattenSwitchState(json: Json): Json =
+    root.state.string
+      .getOption(json)
+      .map(Json.fromString)
+      .getOrElse(json)
+
+  private def flattenAllSwitches: Json => Json =
+    root.each.obj.modify { switchGroup =>
+      switchGroup("switches")
+        .flatMap(_.asObject)
+        .map { switches =>
+          val flattenedSwitches = switches.mapValues(flattenSwitchState)
+          switchGroup
+            .deepMerge(flattenedSwitches)
+            .remove("switches")
+            .remove("description")
+        }
+        .getOrElse(switchGroup)
+    }
+
+  implicit private val customConfig: Configuration = Configuration.default.withDefaults
+  private val switchesEncoder = Encoder[Switches]
+  private val switchesDecoder = Decoder[Switches].prepare(_.withFocus(flattenAllSwitches))
```

Comment:
here we auto-derive a Decoder, but first simplify the json a bit using `prepare`